artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/HierarchicalObjectRepository.java

```diff
@@ -487,13 +487,15 @@ public int compare(final String o1, final String o2) {
             String[] leftSplits = o1.split(quotedDelimiter);
             String[] rightSplits = o2.split(quotedDelimiter);
             for (int i = 0; i < leftSplits.length; i++) {
+               if (i >= rightSplits.length) {
+                  return +1;
```

Comment:
If `i >= rightSplits.length` then we should consider o1/left more specific than 02/right and `return -1`

Reply:
i thought that shorter was more specific but i'll change it.